Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,379 @@ public class ClassificationTests: XCTestCase {
+        ClassificationSpec(source: "\"x\"", kind: .stringLiteral),
+        ClassificationSpec(source: "line", kind: .keyword),
+        ClassificationSpec(source: "1", kind: .integerLiteral),
+      ]
+    )
+  }
+  
+  public func testColorLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as # and identifier")
+    let colorLiteral = "#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)"
+    assertClassification(
+      colorLiteral
+      , expected: [
+        ClassificationSpec(source: colorLiteral, kind: .objectLiteral)
+      ]
+    )
+  }
+  
+  public func testImageLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as a # and an identifier")
+    let imageLiteral = "#imageLiteral(resourceName: \"cloud.png\")"
+    assertClassification(imageLiteral, expected: [ClassificationSpec(source: imageLiteral, kind: .objectLiteral)])
+  }
+  
+  public func testFileLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as # and identifier")
+    let fileLiteral = "#fileLiteral(resourceName: \"cloud.png\")"
+    assertClassification(fileLiteral, expected: [ClassificationSpec(source: fileLiteral, kind: .objectLiteral)])
+  }
+  
+  public func testAttribute() {
+    XCTExpectFailure(
+      """
+      '@available' is classified as @ and a typeIdentifier
+      'iOS' is classified as a typeIdentifier
+      '8.0' and '10.10' are classified as integerLiteral
+      """
+    )
+    assertClassification(
+      """
+      @available(iOS 8.0, OSX 10.10, *)
+      """,
+      expected: [
+        ClassificationSpec(source: "available", kind: .attribute),
+        ClassificationSpec(source: "iOS", kind: .keyword),
+        ClassificationSpec(source: "8.0", kind: .floatingLiteral),
+        ClassificationSpec(source: "OSX", kind: .keyword),
+        ClassificationSpec(source: "10.10", kind: .floatingLiteral),
+        ClassificationSpec(source: "*", kind: .operatorIdentifier),
+      ]
+    )
+  }
+  
+  public func testAttribute2() {
+    XCTExpectFailure(
+      """
+      '@objc' and '@IBOutlet' are classified as @ and a typeIdentifier
+      """
+    )
+    assertClassification(
+      """
+      @objc @IBOutlet var foo: String
+      """
+      , expected: [
+        ClassificationSpec(source: "@objc", kind: .attribute),
+        ClassificationSpec(source: "@IBOutlet", kind: .attribute),
+        ClassificationSpec(source: "var", kind: .keyword),
+        ClassificationSpec(source: "foo", kind: .identifier),
+        ClassificationSpec(source: "String", kind: .typeIdentifier)
+      ]
+    )
+  }
+  
+  public func testIndirectCase() throws {
+    XCTExpectFailure(
+      """
+      'indirect' is classified as an identifier
+      'T' is classified as an identifier
+      'List' is classified as an identifier
+      """
+    )
```

Comment:
This should get correct classification if you wrap it inside an `enum`.

```swift
enum List {
  indirect case cons(T, List)
}
```

And while you’re at it, you can change the case to be lowercase to reflect the naming guidelines which changed around Swift 3.